src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -970,7 +970,7 @@ private void EnforceNotNullIfNotNull(SyntaxNode? syntaxOpt, LocalState state, Im
                         // Parameter '{0}' must have a non-null value when exiting because parameter '{1}' is non-null.
                         Diagnostics.Add(ErrorCode.WRN_ParameterNotNullIfNotNull, location, outputParam.Name, inputParam.Name);
                     }
-                    else
+                    else if (CurrentSymbol is MethodSymbol { IsAsync: false } )
```

Comment:
```suggestion
                    else if (CurrentSymbol is MethodSymbol { IsAsync: false })
```

To (probably) address the formatting issue